packages/editor/src/components/post-taxonomies/flat-term-selector.js

```diff
@@ -34,8 +34,15 @@ const DEFAULT_QUERY = {
 	_fields: 'id,name',
 };
 const MAX_TERMS_SUGGESTIONS = 20;
+
+// Lodash unescape function handles &#39; but not &#039; which may be return in some API requests.
```

Comment:
Which API requests? What other strings are those endpoints returning that we should be aware of?

Reply:
Hi @aduth,
If one issues the following request:
```
wp.apiFetch( {
    path: 'wp/v2/tags?_locale=user',
    method: 'POST',
    data: { name: 'tag2321&#39;aa' },
} ).then( res => {
    console.log( res );
} );
```

The server returns "tag2321&#039;aa" instead of "tag2321&#39;aa" this caused a bug because we did not recognize the strings as equivalent. I'm not sure if there are additional cases it seems like a safer bet would be to use a regex replace to remove zeros at the left from encoding character codes? What do you think?